_posts/2018/2018-05-27-first-and-last-may-not-mean-what-you-think.md

```diff
@@ -0,0 +1,56 @@
+---
+title: Active Record’s first and last may not mean what you think
+description: 'Naming truly is an art'
+layout: article
+category: ruby
+image:
+  base: '2018/first-and-last-may-not-mean-what-you-think'
+  alt: 'Running track'
+  source: 'https://unsplash.com/photos/52p1K0d0euM'
+  credit: 'Austris Augusts'
+---
+
+Each ActiveRecord model comes with the well used ‘get me a single record’ scopes; `.first` & `.last`.
+
+## Instead of…
+
+...using the methods directly on the base ActiveRecord class.
+
+```ruby
+User.first
+User.last
+```
+
+
+## Use…
+
+...them only with ordered scopes. Ideally with named concepts.
+
+### `app/models/user.rb`
+
+```ruby
+class User < ActiveRecord::Base
+  scope :by_created, -> { order(created_at: :asc) }
+  scope :earliest_created, -> { by_created.first }
+  scope :most_recently_created, -> { by_created.last }
+  # ...
+end
+
+User.most_recently_created
+```
+
+
+## But why?
+
+I’ll admit, this is a subtle one. My argument is mostly about making the code more explicit.
```

Comment:
I think there's a better (subtly different) way to say this sentence, but maybe not: My argument is mostly about making the code more explicit.